libs/common/result.hpp

```diff
@@ -114,13 +125,45 @@ namespace iroha {
      * result
      */
     template <typename T, typename E, typename Transform>
-    constexpr auto operator|(Result<T, E> r, Transform &&f)
-        -> decltype(f(std::declval<T>())) {
+    constexpr auto operator|(Result<T, E> r, Transform &&f) ->
+        typename std::enable_if<
+            not std::is_same<decltype(f(std::declval<T>())), void>::value,
+            decltype(f(std::declval<T>()))>::type {
       using return_type = decltype(f(std::declval<T>()));
       return r.match(
           [&f](const Value<T> &v) { return f(v.value); },
           [](const Error<E> &e) { return return_type(makeError(e.error)); });
     }
+
+    /**
+     * Bind operator overload for functions which return void. Does nothing in
+     * error case, or executes function in a value case
+     * @param f function which returns void
+     */
+    template <typename T, typename E, typename Procedure>
+    constexpr auto operator|(Result<T, E> r, Procedure &&f) ->
+        typename std::enable_if<
+            std::is_same<decltype(f(std::declval<T>())), void>::value>::type {
+      return r.match([&f](const Value<T> &v) { f(v.value); },
+                     [](const Error<E> &e) {});
```

Comment:
What it the point of using result if u are not interesting in the error value? Maybe in this case required to use value type instead?

Reply:
Actually, I made this function as a temporary solution for some issue, and now it is unused, so it should be removed. Good catch!